core/src/vmem.rs

```diff
@@ -55,7 +55,7 @@ impl Debug for VirtualMemory {
 
 impl VirtualMemory {
     /// The maximum allocation size for a `wasmi` virtual memory.
-    const MAX_ALLOCATION_SIZE: usize = u32::MAX as usize;
+    const MAX_ALLOCATION_SIZE: usize = u32::MAX as usize + 1;
```

Comment:
This breaks on 32-bit architectures since `usize == u32`. Therefore we need to use `u64` here to cover for this case OR make a distinction between 32-bit and 64-bit architectures were 32-bit architectures can only allocate at most 2GB.

For newer `wasmi` versions I already have a draft PR in the pipeline:
https://github.com/paritytech/wasmi/pull/449

Reply:
Good point :facepalm:

Reply:
Fixes